qualtran/bloqs/block_encoding/chebyshev_polynomial.py

```diff
@@ -129,23 +128,23 @@ def signal_state(self) -> PrepareOracle:
 
     @cached_property
     def reflection_bloq(self):
-        return GlobalPhase(exponent=1).controlled(
-            ctrl_spec=CtrlSpec(qdtypes=QAny(self.ancilla_bitsize), cvs=0)
+        return ReflectionUsingPrepare.reflection_around_zero(
+            bitsizes=(self.ancilla_bitsize,), global_phase=-1
         )
 
     def build_composite_bloq(self, bb: BloqBuilder, **soqs: SoquetT) -> Dict[str, SoquetT]:
         if is_symbolic(self.ancilla_bitsize):
             raise DecomposeTypeError(f"Cannot decompose symbolic {self=}")
         for _ in range(self.order // 2):
             if self.ancilla_bitsize > 0:
-                soqs["ancilla"] = bb.add(self.reflection_bloq, ctrl=soqs["ancilla"])
+                soqs["ancilla"] = bb.add(self.reflection_bloq, reg0_=soqs["ancilla"])
```

Comment:
This seems fragile, would it break if we used a different prepare bloq? Or rather, is the prepare bloq configurable by the user or always picked to be the same one?

Reply:
What do you mean by different prepare bloq? the `reg0_` name is dictated by the `ReflectionUsingPrepare`. The reflection bloq in `ChebyshevPolynomial` is not user-configurable.

Reply:
Ah nvm, I thought this was reflecting around the signal state.